src/troute-routing/troute/routing/fast_reach/mc_reach.pyx

```diff
@@ -621,6 +624,21 @@ cpdef object compute_network_structured(
                     lastobs_values[gage_i],
                     gage_i == da_check_gage,
                 )
+                 # Ensure we're within bounds before accessing the arrays
+                if timestep < usgs_values.shape[1]:
```

Comment:
I think it's better to use gage_maxtimestep instead of usgs_values.shape[1] for clarity.

Reply:
Done